### DISCUSSION TITLE
Create object from Zod object with defaults but that will not pass its validation

### DISCUSSION THREAD

**Discussion by andredewaard** [colinhacks/zod#2487] [Ideas]
Is it possible to create an object from a Zod object with defaults BUT that will not pass its validation? 
So for example i have
```ts
const EmailForm = z.object({
  email: z.string().email().default('')
})

const form = EmailForm.parse({}) // This will create the object as { email: '' }
```
But when i want to validate that the object has an correct email, This should fail
```ts
EmailForm.parse(form) // Throw error since form.email is not a correct email

form.email = 'test@test.com'
EmailForm.parse(form) // Passes
```

**Comment by mtsknn** [comment]
`z.string().email().default('')` indeed seems to work only with valid emails (wanted to confirm this because I wasn't sure if I had interpreted your question correctly):

```ts
const EmailForm = z.object({
  email: z.string().email().default(''),
})

EmailForm.parse({}) //=> ZodError: Invalid email
EmailForm.parse({ email: undefined }) //=> ZodError: Invalid email
EmailForm.parse({ email: 'gibberish' }) //=> ZodError: Invalid email
EmailForm.parse({ email: 'test@test.com' }) //=> { email: "test@test.com" }
```

---

A union works and is quite legible ("accept an email string, or accept `undefined` and transform it to an empty string"):

```ts
const EmailForm = z.object({
  email: z.union([z.string().email(), z.undefined().transform(() => '')]),
})

EmailForm.parse({}) //=> { email: "" }
EmailForm.parse({ email: undefined }) //=> { email: "" }
EmailForm.parse({ email: 'test@test.com' }) //=> { email: "test@test.com" }
EmailForm.parse({ email: 'gibberish' }) //=> ZodError: Invalid email
```

---

Things that I tried but that didn't work:

- `z.string().email().optional().default('')`: still getting `ZodError: Invalid email` from `EmailForm.parse({})`.
- `z.undefined().default('')` (i.e. using this in the union instead of `z.undefined().transform(() => '')`): TypeScript complains that the `default`'s param should be of type `never` or `() => never`.
